Enable Refresh Lock Request

birdstorm opened the pull request:
Provide `RefreshLockRequest` for long-lasting transactions. Users can now use a lock with short TTL and refresh it at ease.

Merge `release-3.0` branch to `master` so that we will be using `master` branch of `kvproto` in `TiKV` instead of `release-3.0`.

MyonKeminta commented on proto/tikvpb.proto:
```diff
@@ -70,7 +71,7 @@ service Tikv {
 message BatchCommandsRequest {
```
I think it's better to add it to `BatchCommands`, so that it can be batched with other requests to save RPC.

youjiali1995 commented on proto/kvrpcpb.proto:
```diff
@@ -295,6 +295,20 @@ message ResolveLockResponse {
+    uint64 start_version = 2;
+    bytes key = 3;
+    uint64 ttl = 4;
+    uint64 txn_size = 5;
```
What the use of txn_size?

birdstorm commented:
@MyonKeminta @youjiali1995 I have moved this PR to #414 so that it will base on release-3.0. Comments are now addressed in #414.

This PR will be closed.

I must say that it is rather strange that TiKV’s `master` now bases on kvproto’s `release-3.0` branch.

MyonKeminta commented:
I think this PR should continue. We will switch TiKV's kvproto back to master in the future.
